pkg/kubelet/eviction/threshold_notifier_linux.go

```diff
@@ -58,7 +58,7 @@ func NewCgroupNotifier(path, attribute string, threshold int64) (CgroupNotifier,
 		return nil, err
 	}
 	defer unix.Close(controlfd)
-	eventfd, err = unix.Eventfd(0, unix.EFD_CLOEXEC)
+	eventfd, err = unix.Eventfd(0, unix.EFD_NONBLOCK|unix.EFD_CLOEXEC)
```

Comment:
looking at code and at manpages, this is the only change that makes me think, I'll need to dig into the history and into the codebase to learn why we didn't introduce this flag earlier.

Reply:
Sure: if you could elaborate what's the benefit of this change - just this change, just adding this flag in the context of this code, that will make things faster. At glance (still digging), considering how the code behaves in `wait()` (https://github.com/kubernetes/kubernetes/blob/v1.20.4/pkg/kubelet/eviction/threshold_notifier_linux.go#L141) I don't see yet an obvious benefit.

Reply:
Any updates on this?

Reply:
> Sure: if you could elaborate what's the benefit of this change - just this change, just adding this flag in the context of this code, that will make things faster. At glance (still digging), considering how the code behaves in `wait()` (https://github.com/kubernetes/kubernetes/blob/v1.20.4/pkg/kubelet/eviction/threshold_notifier_linux.go#L141) I don't see yet an obvious benefit.

It's more like a convention flag, it makes the `write` to the inside uint64_t counter of `eventfd` never block, of course, in theory.

Reply:
So there is not a performance gain nor a bugfix rationale for this change? I'd remove from this set then.

Reply:
Actually, I think `unix.EFD_CLOEXEC` is also a convention flag here without actual impacts?